.github/workflows/build-and-publish.yml

```diff
@@ -40,14 +40,15 @@ jobs:
       - name: Get commit hash
         id: get-commit-hash
         run: |
-          COMMIT_HASH=$(git rev-parse ${{ inputs.ref }})
+          # HEAD should be the same as inputs.ref since we checked out inputs.ref
+          COMMIT_HASH=$(git rev-parse HEAD)
           echo "Commit hash: $COMMIT_HASH"
           echo "commit_hash=$COMMIT_HASH" >> "$GITHUB_OUTPUT"
   build-and-publish:
     name: Build and publish
     runs-on: ubuntu-latest
     needs: get-commit-hash
-    concurrency: ${{ github.workflow }}-${{ needs.get-commit-hash.outputs.commit_hash }}
+    concurrency: ${{ github.workflow }}-${{ inputs.app_name }}-${{ needs.get-commit-hash.outputs.commit_hash }}
```

Comment:
Add app_name to the concurrency group so that multiple apps can build at the same time